magento/magento2
# 31353 watermark images graphql

### Description (*)

I changed \Magento\Catalog\Model\Product\Image class to add logic to getUrl() method.
I added checks if cached image files doesn't exist then create new cache file with watermark. 
in \Magento\Catalog\Model\View\Asset\Image class I added 2 public methods to get media format for url and to get cached file path. This methods I used in Image class fro checks


### Fixed Issues (if relevant)
1. Fixes magento/magento2#31353

### Manual testing scenarios (*)
1. Add config for watermarks for images in design configuration.
2. Create query for GraphQl to get product images.
3. Check if product's images have watermark.

### Contribution checklist (*)
